addons/website_event_booth/controllers/event_booth.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import werkzeug
+from werkzeug.exceptions import Forbidden, NotFound
+
+from odoo import http, _
+from odoo.http import request
+from odoo.addons.website_event.controllers.main import WebsiteEventController
+
+
+class WebsiteEventBoothController(WebsiteEventController):
+
+    @http.route('/event/<model("event.event"):event>/booth', type='http', auth='public', website=True, sitemap=False)
+    def event_booth_main(self, event):
+        event_sudo = event.sudo()
+        values = {
+            'event': event,
+            'event_booth_ids': event_sudo.event_booth_ids,
+            'available_booth_category_ids': event_sudo.event_available_booth_category_ids,
+            'main_object': event,
+        }
+        return request.render('website_event_booth.event_booth_registration', values)
+
+    @http.route('/event/<model("event.event"):event>/booth/register', type='http', auth='public', methods=['POST'], website=True)
+    def event_booth_register(self, event, **kwargs):
+        event_booth_ids = request.httprequest.form.getlist('event_booth_ids')
+
+        return request.redirect(('/event/%s/booth/continue?' % event.id) + werkzeug.urls.url_encode({
+            'booths': ','.join(event_booth_ids),
+            'booth_category': kwargs.get('booth_category_id'),
+        }))
+
+    @http.route('/event/<model("event.event"):event>/booth/continue', type='http', auth='public', methods=['GET'], website=True)
+    def event_booth_contact_form(self, event, booths, booth_category):
+        if not booths or not booth_category:
+            raise NotFound()
+
+        booth_category_id = request.env['event.booth.category'].sudo().browse(int(booth_category))
+        default_contact = {}
+        if not request.env.user._is_public():
+            default_contact = {
+                'name': request.env.user.partner_id.name,
+                'email': request.env.user.partner_id.email,
+                'phone': request.env.user.partner_id.phone,
+                'mobile': request.env.user.partner_id.mobile,
+            }
+        else:
+            visitor = request.env['website.visitor']._get_visitor_from_request()
+            if visitor.email:
+                default_contact = {
+                    'name': visitor.name,
+                    'email': visitor.email,
+                    'mobile': visitor.mobile,
+                }
+        return request.render(
+            'website_event_booth.event_booth_registration_details',
+            {'event': event.sudo(),
+             'default_contact': default_contact,
+             'booth_category_id': booth_category_id,
+             'event_booth_ids': booths,
+            }
+        )
+
+    @http.route('/event/<model("event.event"):event>/booth/confirm', type='http', auth='public', methods=['POST'], website=True)
+    def event_booth_registration_confirm(self, event, **kwargs):
```

Comment:
I think some defensive code could be added here: if I am not wrong, we need an event_booth_ids (that contains at least one valid ID), and booth_category_id. Moreover all booths should belong to the same category (otherwise you will have issues in sale override). Am I right ? Could be maybe add those constraints here so that we are sure data is valid afterwards ?